sdk/storage/azblob/bloberror/error_codes.go

```diff
@@ -149,3 +149,7 @@ const (
 	UnsupportedQueryParameter                         Code = "UnsupportedQueryParameter"
 	UnsupportedXMLNode                                Code = "UnsupportedXmlNode"
 )
+
+var (
+	NoSharedKeyCredential = errors.New("SAS can only be signed with a SharedKeyCredential")
```

Comment:
You'll want to add a doc string explaining when this can be returned.

Reply:
I'm also wondering if this should be named `MissingSharedKeyCredential`.  While "No" is shorter, I feel like it states, "you can't use SharedKeyCredential" instead of "a SharedKeyCredential is required but absent".

Reply:
That's a really good point, I updated the name to reflect this!